File: files/CSEC 24.11.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge Nov 24th.xlsx"
input = read_excel(path, range = "B2:B14")

result = input %>%
  mutate(cond_form = ifelse(str_detect(Items, "\\D+ \\d+ \\D+ \\d+", negate = T), "Yes", ""))

File: files/2024-11-24/CSEC 24.11.py
import pandas as pd
import re

path = "Excel Challenge Nov 24th.xlsx"
input_data = pd.read_excel(path, usecols="B", skiprows=1, nrows=13)

pattern = r"\D+ \d+ \D+ \d+"
result = input_data.assign(cond_form=input_data["Items"].map(lambda x: "" if re.fullmatch(pattern, str(x)) else "Yes"))
print(result)
